java/adapter/avro/src/main/java/org/apache/arrow/consumers/AvroStringConsumer.java

```diff
@@ -43,12 +44,15 @@ public AvroStringConsumer(VarCharVector vector) {
   @Override
   public void consume(Decoder decoder) throws IOException {
     VarCharHolder holder = new VarCharHolder();
-    ByteBuffer byteBuffer = decoder.readBytes(null);
+
+    //cacheBuffer is initialized null and create in the first consume,
+    //if its capacity < size to read, decoder will create a new one with new capacity.
+    cacheBuffer = decoder.readBytes(cacheBuffer);
 
     holder.start = 0;
-    holder.end = byteBuffer.capacity();
-    holder.buffer = vector.getAllocator().buffer(byteBuffer.capacity());
-    holder.buffer.setBytes(0, byteBuffer);
+    holder.end = cacheBuffer.limit();
+    holder.buffer = vector.getAllocator().buffer(cacheBuffer.limit());
+    holder.buffer.setBytes(0, cacheBuffer, 0, cacheBuffer.limit());
```

Comment:
it would be nice to eliminate duplication here with VarBinary if at all possible

Reply:
They have different type holder, seems hard to unify together?

Reply:
Hmm, we might want to go back to direct vector access or make a common some sort of common interface, this can probably be done as a follow-up (please make a JIRA so we don't forget).

Reply:
Sure, thanks!